rclcpp/include/rclcpp/create_publisher.hpp

```diff
@@ -25,12 +25,14 @@
 namespace rclcpp
 {
 
-template<typename MessageT, typename AllocatorT, typename PublisherT>
+template<typename MessageT, typename EventCallbackT, typename AllocatorT, typename PublisherT>
 std::shared_ptr<PublisherT>
 create_publisher(
   rclcpp::node_interfaces::NodeTopicsInterface * node_topics,
   const std::string & topic_name,
   const rmw_qos_profile_t & qos_profile,
+  EventCallbackT && event_callback,
+  rclcpp::callback_group::CallbackGroup::SharedPtr group,
```

Comment:
This might be a good time to implement something like the `rclcpp::NodeOptions` class, but for publishers, subscriptions, etc... It will be especially helpful when/if we allow for one callback per type of event.

Reply:
We will have a separate PR for introducing `rclcpp::SubscriptionOptions` and `rclcpp::PublisherOptions`.